zebra/zebra_mpls.c

```diff
@@ -1103,6 +1086,37 @@ static void *lsp_alloc(void *p)
 	return ((void *)lsp);
 }
 
+/*
+ * Dtor for an LSP: remove from ile hash, release any internal allocations,
+ * free LSP object.
+ */
+static void lsp_dtor(struct hash *lsp_table, zebra_lsp_t **plsp)
+{
+	zebra_lsp_t *lsp;
+	zebra_nhlfe_t *nhlfe, *nhlfe_next;
+
+	if (plsp == NULL || *plsp == NULL)
+		return;
+
+	lsp = *plsp;
+
+	if (IS_ZEBRA_DEBUG_MPLS)
+		zlog_debug("Free LSP in-label %u flags 0x%x",
+			   lsp->ile.in_label, lsp->flags);
+
+	/* Free nhlfes, if any. */
+	for (nhlfe = lsp->nhlfe_list; nhlfe; nhlfe = nhlfe_next) {
```

Comment:
Seems like a fairly large memory leak if anyone was using lsp's? right?  Should this be backported to 7.3 and 7.2?

Reply:
At the moment, this is only _called_ when there are no more nhlfes in the list; just wanted to be a bit more general. So I don't think there's an existing issue with previous releases.